sdk/test-utils/perfstress/src/tests.ts

```diff
@@ -48,6 +50,12 @@ export abstract class PerfStressTest<TOptions = {}> {
   public cleanup?(): void | Promise<void>;
 
   public async runAsync?(abortSignal?: AbortSignalLike): Promise<void>;
+
+  public getRecordingClient(): RecordingHttpClient {
```

Comment:
```suggestion
  public getHttpClient(): HttpClient {
```

It's important that we only use `TestProxyHttpClient` when the `test-proxy` option is set.  When `test-proxy` is **not** set,  we should use the default codepath to avoid influencing the live test results.

In .NET, this is implemented by only modifying the `ClientOptions` object if the option is set:

https://github.com/Azure/azure-sdk-for-net/pull/22621/files#diff-427d3566282e8c258aa70a4b5d4eddbbfc5cf250b38a8e0dcfcb7b43126b9f18R64-R70

The JS implementation will be different, since clients directly consume the `HttpClient` rather than a `ClientOptions` object.  One approach is that `getHttpClient()` could return `DefaultHttpClient` if the option is not set, and `TestProxyHttpClient` if the option is set.

Reply:
Right now, we only call the recorder methods when test-proxy option is passed.
![image](https://user-images.githubusercontent.com/10452642/127561067-7dd516be-066d-49c7-9afa-03cbb95b251e.png)
And sendRequest is modified only if the mode is record or playback.
So, live is unaffected.

I can still add the check and return DefaultHttpClient. 🙌